cibuildwheel/architecture.py

```diff
@@ -183,14 +176,32 @@ def all_archs(platform: PlatformName) -> "set[Architecture]":
 
     @staticmethod
     def bitness_archs(platform: PlatformName, bitness: Literal["64", "32"]) -> "set[Architecture]":
-        archs_32 = {Architecture.i686, Architecture.x86, Architecture.armv7l}
-        auto_archs = Architecture.auto_archs(platform)
-
-        if bitness == "64":
-            return auto_archs - archs_32
-        if bitness == "32":
-            return auto_archs & archs_32
-        typing.assert_never(bitness)
+        # This map maps 64-bit architectures to their 32-bit equivalents.
+        archs_map = {
+            Architecture.x86_64: Architecture.i686,
+            Architecture.AMD64: Architecture.x86,
+            Architecture.aarch64: Architecture.armv7l,
+        }
+        native_arch = Architecture.native_arch(platform)
+
+        if native_arch is None:
+            return set()  # can't build anything on this platform
+
```

Comment:
In the current state of the PR, we'll have `wasm32` in `auto64` but not in `auto32`. I'm thinking for correctness we shouldn't return `wasm32` in `auto64`, but should return it in `auto32`.

```suggestion

        if native_arch == Architecture.wasm32:
            return native_arch if bitness == "32" else set()

```